master/pkg/stream/stream.go

```diff
@@ -105,42 +105,36 @@ func NewSubscription[T Msg](
 	streamer *Streamer,
 	publisher *Publisher[T],
 	permFilter func(T) bool,
+	filterFn func(T) bool,
 ) Subscription[T] {
 	return Subscription[T]{
 		Streamer:         streamer,
 		Publisher:        publisher,
 		permissionFilter: permFilter,
+		filter:           filterFn,
 	}
 }
 
-// Configure updates a Subscription's filters and updates the associated Publisher
-// in the event of creating or deleting a Subscription.
-func (s *Subscription[T]) Configure(filter func(T) bool) {
-	if filter == nil && s.filter == nil {
-		// no change, no synchronization needed
-		return
-	}
-	// Changes must be synchronized with our respective publisher.
+// Unsubscribe removes the subscription from its Publisher.
+func (s *Subscription[T]) Unsubscribe() {
```

Comment:
it's weird that ".Unsubscribe" is the inverse operation to ".Register", can we pick either register/unregister, or subscribe/unsubscribe?

Reply:
Changed to `.Deregister()` and `.DeregisterAll()`